core/src/main/java/com/github/games647/fastlogin/core/shared/FloodgateManagement.java

```diff
@@ -0,0 +1,131 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.github.games647.fastlogin.core.shared;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.geysermc.floodgate.api.player.FloodgatePlayer;
+
+import com.github.games647.craftapi.model.Profile;
+import com.github.games647.craftapi.resolver.RateLimitException;
+import com.github.games647.fastlogin.core.StoredProfile;
+import com.github.games647.fastlogin.core.hooks.AuthPlugin;
+
+public abstract class FloodgateManagement<P extends C, C, L extends LoginSession, T extends PlatformPlugin<C>>
+        implements Runnable {
+
+    protected final FastLoginCore<P, C, T> core;
+    protected final P player;
+    private final FloodgatePlayer floodgatePlayer;
+    private final String username;
+
+    //variables initialized through run() and accesses by subclasss
+    protected boolean isRegistered;
+    protected StoredProfile profile;
+    protected String autoLoginFloodgate;
+    protected boolean isLinked;
+    protected boolean performLogin; //will be set to ture if core#run() wasn't interrupted by a return;
+
+    public FloodgateManagement(FastLoginCore<P, C, T> core, P player, FloodgatePlayer floodgatePlayer) {
+        this.core = core;
+        this.player = player;
+        this.floodgatePlayer = floodgatePlayer;
+        this.username = getName(player);
+    }
+
+    @Override
+    public void run() {
+        core.getPlugin().getLog().info(
+                "Player {} is connecting through Geyser Floodgate.",
+                username);
+
+        // check if the Bedrock player is linked to a Java account 
+        isLinked = floodgatePlayer.getLinkedPlayer() != null;
+        AuthPlugin<P> authPlugin = core.getAuthPluginHook();
+
+        autoLoginFloodgate = core.getConfig().get("autoLoginFloodgate").toString().toLowerCase();
+        String autoRegisterFloodgate = core.getConfig().get("autoRegisterFloodgate").toString().toLowerCase();
+        String allowNameConflict = core.getConfig().get("allowFloodgateNameConflict").toString().toLowerCase();
+        
+        try {
+            isRegistered = authPlugin.isRegistered(username);
+        } catch (Exception e) {
+            core.getPlugin().getLog().error(
+                    "An error has occured while checking if player {} is registered",
+                    username);
+            return;
+        }
+
+        //decide if checks should be made for conflicting Java player names
+        if (!isLinked //linked players have the same name as their Java profile
```

Comment:
I realized now this check is too complicated to read. Maybe you should extract it into a method or multiple program flows.